api/v1beta1/types.go

```diff
@@ -381,6 +391,30 @@ func (node *OscNode) SetVolumeDefaultValue() {
 	}
 }
 
+// SetVolumeDefaultValue set the Volume default values from volume configuration
+func (node *OscNode) SetKeyPairDefaultValue() {
+	bitSize := 2048
+	if len(node.KeyPair.PublicKey) == 0 {
+		privateKey, err := rsa.GenerateKey(rand.Reader, bitSize)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = privateKey.Validate()
+		if err != nil {
+			log.Fatal(err)
+		}
+		publicRsaKey, err := ssh.NewPublicKey(privateKey)
+		if err != nil {
+			log.Fatal(err)
+		}
```

Comment:
I would only rely on `crypto` package instead of `x/crypto/ssh`.

There is a method called `Public()` in `crypto/rsa` to get the public key. Is there a reason not to use this method ?